r-pkg/R/elasticsearch_eda_funs.R

```diff
@@ -114,3 +114,76 @@ get_counts <- function(field
     
 }
 
+#' @title Get the names and data types of the indexed fields in an index
+#' @name get_fields
+#' @description For a given Elasticsearch index, return the mapping from field name
+#'              to data type for all indexed fields.
+#' @importFrom data.table := as.data.table setnames
+#' @importFrom futile.logger flog.fatal
+#' @importFrom httr GET content stop_for_status
+#' @importFrom stringr str_detect str_split_fixed str_replace
+#' @param es_host A string identifying an Elasticsearch host. This should be of
+#'                the form \code{[transfer_protocol][hostname]:[port]}. For example,
+#'                \code{'http://myindex.thing.com:9200'}.
+#' @param es_indexes A character vector that contains the names of indexes for
+#'                   which to get mappings. Default, is \code{'_all'}, which means
+#'                   get the mapping for all indexes. Names of indexes can be
+#'                   treated as regular expressions.
+#' @export
+#' @return A data.table containing four columns: index, type, field, and datatype
+#' @examples \dontrun{
+#' # get the mapping for all types in the ticket_sales and customers indexes
+#' mappingDT <- retrieve_mapping(es_host = "http://es.custdb.mycompany.com:9200"
+#'                               , es_indexes = c("ticket_sales", "indexes"))
+#' }
+get_fields <- function(es_host
+                       , es_indexes = '_all'
+) {
+    
+    # Input checking
+    url <- uptasticsearch:::.ValidateAndFormatHost(es_host)
```

Comment:
You don't need to do this. `.ValidateAndFormatHost` from this package is guaranteed to be found first on the search path. Remove the `:::`